From 63861c039c4fa8ee2882a349153f31f6f87a5ac9 Mon Sep 17 00:00:00 2001
From: Guillaume Rousse <guillomovitch@gmail.com>
Date: Wed, 22 Feb 2017 18:24:30 +0100
Subject: [PATCH] restore out-of-the-box installation logic

- move setup.pm to library directory
- ensure substitutions are performed at installation time on this file
- do not perform useless substituions on executables
---
 Makefile.PL                       | 12 ++++++------
 bin/fusioninventory-agent         |  2 +-
 bin/fusioninventory-esx           |  2 +-
 bin/fusioninventory-inventory     |  2 +-
 bin/fusioninventory-netdiscovery  |  2 +-
 bin/fusioninventory-netinventory  |  2 +-
 bin/fusioninventory-wakeonlan     |  2 +-
 bin/fusioninventory-win32-service |  2 +-
 bin/setup.pm                      | 37 -------------------------------------
 lib/setup.pm                      | 16 ++++++++++++++++
 10 files changed, 29 insertions(+), 50 deletions(-)
 delete mode 100644 bin/setup.pm
 create mode 100644 lib/setup.pm

diff --git a/Makefile.PL b/Makefile.PL
index e28e537..f8ffbe8 100644
--- a/Makefile.PL
+++ b/Makefile.PL
@@ -232,8 +232,12 @@ data_install :
 	install -m 644 share/html/* $(DESTDIR)$(DATADIR)/html
 
 setup_install :
-	install -d -m 755 $(DESTDIR)$(DATADIR)/lib
-	install -m 644 bin/setup.pm $(DESTDIR)$(DATADIR)/lib/
+	$(ABSPERLRUN) -pi \
+	-e 's|confdir => .*|confdir => "$(SYSCONFDIR)",|;' \
+	-e 's|datadir => .*|datadir => "$(DATADIR)",|;' \
+	-e 's|libdir  => .*|libdir  => "$(DATADIR)/lib",|;' \
+	-e 's|vardir  => .*|vardir  => "$(LOCALSTATEDIR)",|' \
+	$(DESTDIR)$(DATADIR)/lib/setup.pm
 EOF
     return $install;
 }
@@ -250,10 +254,6 @@ sub installbin {
     $installbin .= <<'EOF';
 FINALIZE = $(ABSPERLRUN) -pi \
     -e 's|use lib .*|use lib "$(DATADIR)/lib";|;' \
-    -e 's|confdir => .*|confdir => "$(SYSCONFDIR)",|;' \
-    -e 's|datadir => .*|datadir => "$(DATADIR)",|;' \
-    -e 's|libdir  => .*|libdir  => "$(DATADIR)/lib",|;' \
-    -e 's|vardir  => .*|vardir  => "$(LOCALSTATEDIR)",|' \
     --
 EOF
     return $installbin;
diff --git a/bin/fusioninventory-agent b/bin/fusioninventory-agent
index 02ae72e..95b872b 100755
--- a/bin/fusioninventory-agent
+++ b/bin/fusioninventory-agent
@@ -3,7 +3,7 @@
 use strict;
 use warnings;
 
-use lib './bin';
+use lib './lib';
 use setup;
 
 use English qw(-no_match_vars);
diff --git a/bin/fusioninventory-esx b/bin/fusioninventory-esx
index 037bed2..220e30c 100755
--- a/bin/fusioninventory-esx
+++ b/bin/fusioninventory-esx
@@ -3,7 +3,7 @@
 use strict;
 use warnings;
 
-use lib './bin';
+use lib './lib';
 use setup;
 
 use English qw(-no_match_vars) ;
diff --git a/bin/fusioninventory-inventory b/bin/fusioninventory-inventory
index 94d7e1d..18a8521 100755
--- a/bin/fusioninventory-inventory
+++ b/bin/fusioninventory-inventory
@@ -3,7 +3,7 @@
 use strict;
 use warnings;
 
-use lib './bin';
+use lib './lib';
 use setup;
 
 use English qw(-no_match_vars);
diff --git a/bin/fusioninventory-netdiscovery b/bin/fusioninventory-netdiscovery
index 0e2b0c7..52cc2e2 100755
--- a/bin/fusioninventory-netdiscovery
+++ b/bin/fusioninventory-netdiscovery
@@ -3,7 +3,7 @@
 use strict;
 use warnings;
 
-use lib './bin';
+use lib './lib';
 use setup;
 
 use English qw(-no_match_vars);
diff --git a/bin/fusioninventory-netinventory b/bin/fusioninventory-netinventory
index 3562b6b..518c3a7 100755
--- a/bin/fusioninventory-netinventory
+++ b/bin/fusioninventory-netinventory
@@ -3,7 +3,7 @@
 use strict;
 use warnings;
 
-use lib './bin';
+use lib './lib';
 use setup;
 
 use English qw(-no_match_vars);
diff --git a/bin/fusioninventory-wakeonlan b/bin/fusioninventory-wakeonlan
index 04c3e3c..6ed4215 100755
--- a/bin/fusioninventory-wakeonlan
+++ b/bin/fusioninventory-wakeonlan
@@ -3,7 +3,7 @@
 use strict;
 use warnings;
 
-use lib './bin';
+use lib './lib';
 use setup;
 
 use English qw(-no_match_vars);
diff --git a/bin/fusioninventory-win32-service b/bin/fusioninventory-win32-service
index 27b58b9..9ad9da9 100644
--- a/bin/fusioninventory-win32-service
+++ b/bin/fusioninventory-win32-service
@@ -3,7 +3,7 @@
 use strict;
 use warnings;
 
-use lib './bin';
+use lib './lib';
 use setup;
 
 use threads;
diff --git a/bin/setup.pm b/bin/setup.pm
deleted file mode 100644
index 3f92279..0000000
--- a/bin/setup.pm
+++ /dev/null
@@ -1,37 +0,0 @@
-package setup;
-
-use strict;
-use warnings;
-use base qw(Exporter);
-
-our @EXPORT = ('%setup');
-
-our %setup;
-
-# From here we can setup @INC so any needed perl module can be found. We add
-# as many 'use lib' directive as needed
-# We could also define '%setup' hash while useful
-
-# Here is a sample working from sources directory or its bin subfolder
-if (-d 'lib') {
-    use lib './lib' ;
-
-    %setup = (
-        confdir => './etc',
-        datadir => './share',
-        libdir  => './lib',
-        vardir  => './var',
-    );
-
-} elsif (-d '../lib') {
-    use lib '../lib';
-
-    %setup = (
-        confdir => '../etc',
-        datadir => '../share',
-        libdir  => '../lib',
-        vardir  => '../var',
-    );
-}
-
-1;
diff --git a/lib/setup.pm b/lib/setup.pm
new file mode 100644
index 0000000..a882f7f
--- /dev/null
+++ b/lib/setup.pm
@@ -0,0 +1,16 @@
+package setup;
+
+use strict;
+use warnings;
+use base qw(Exporter);
+
+our @EXPORT = ('%setup');
+
+our %setup = (
+    confdir => './etc',
+    datadir => './share',
+    libdir  => './lib',
+    vardir  => './var',
+);
+
+1;
